### DISCUSSION TITLE
Webportal

### DISCUSSION THREAD

**Discussion by maikheinrich** [spali/go-rscp#11] [Q&A]
Hi Spali,

ist es auch möglich mit deiner go-rscp API über das Webportal von e3dc die Daten abzufragen ? Ich habe eine weitere e3dc Anlage mit meinem Webportal verknüpft und möchte von dieser ebenfalls Daten abfragen.

LG Maik

**Comment by spali** [comment]
Nicht das ich wüsste. Ich gehe nicht davon aus, dass das Webportal rscp spricht.

**Comment by SaschaHenning** [comment]
Kannst du eine VPN zu dem Netz aufbauen, in dem die andere Anlage ist? Wenn du dann per IP Zugriff hast sollte der RSCP ja evtl. funktionieren. 
Um Daten aus dem WebPortal auszulesen, müsstest du ja eher die Seite parsen / den Websocket selbst aufbauen.

**Reply by maikheinrich** [reply]
> Kannst du eine VPN zu dem Netz aufbauen, in dem die andere Anlage ist? Wenn du dann per IP Zugriff hast sollte der RSCP ja evtl. funktionieren.
> Um Daten aus dem WebPortal auszulesen, müsstest du ja eher die Seite parsen / den Websocket selbst aufbauen.

Hallo Sascha,

nein VPN ist leider nicht möglich. Es gibt zwar ein Site2Site VPN mit unterschiedlichen Netzwerkadressen. Aber deer E3DC beantwortet anfragen nur an Empfänger im eigenen Netz.

Das E3DC Webportal unterstützt RSCP. Zumindest kann z. B. mit diesem Projekt über das Webportal auf die Geräte zugegriffen werden. https://github.com/rxhan/RSCPGui

LG Maik

**Reply by spali** [reply]
Das kann man mit der entsprechenden Firewall auf der E3DC Seite lösen:
Hier ein Beispiel wie ich es mit OPNsense nutze.
Das ist die Einstellung auf dem Firewall auf der E3DC Seite:
![image](https://user-images.githubusercontent.com/1839708/152602544-b184dba1-40e0-4eb2-92ea-83d385a23948.png)
`Net_Home_LAN` ist ein Alias für das Ursprungsnetzwerk, von wo aus du die Anfrage machst. `10.1.0.92` Ist in dem Fall die IP vom E3DC.
`NAT Address` sagt in dem Fall der Firewall das die Packete ausgehend seine lokale IP als Absender haben sollen.
Konkret, münzt damit die Firewall den Absender der Pakete soweit um, dass alles was von `Net_Home_LAN` Subnetz kommt und an `10.1.0.92/32` auf TCP Port 5033 geht, mit der LAN IP des Firewalls als Absender an die E3DC geht.
Somit kann die E3DC den unterschied nicht mehr erkennen ;)

Im Prinzip das gleiche was jeder Router mit NAT für Internet macht. In dem Fall halt einfach nur für spezifische Pakete welche in dem Fall vom VPN aus einem anderen Netz kommen.
